Makefile

```diff
@@ -0,0 +1,64 @@
+LOCAL_IMAGE = acorn_docker:1.0
+REMOTE_IMAGE = merlinran/acorn_docker
+
+.PHONY: list
+list:
+	@echo "Welcome! You may want to try one of the following:\n---"; \
+	grep "^.PHONY:" Makefile | grep "#" | cut -d ":" -f 2- | sed "s/\w*#/\t#/g" | sed "s/^/make/"
+
+.PHONY: simulation # Run the vehicle and server containers in simulation mode.
+simulation: docker-image
```

Comment:
Here `docker-image` is a dependency of `simulation`. Before running the body of the rule, `make` first makes sure `docker-image` is up-to-date, by recursively checking the rule at [L59](https://github.com/Twisted-Fields/acorn-precision-farming-rover/pull/17/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R59) and running the body if needed.
In this case, the dependency chain is `simulation` <- `docker-image` <- `Dockerfile` <- `vehicle/requirements.txt` and `server/requirements.txt`. As long as the upstream target is newer than the downstream target, the rule body will be run. It would be very difficult to express such relationship in plain shell or Python.